diff -uNr apache-activemq-5.13.1/bin/linux-x86-32/activemq apache-activemq-5.13.1.b/bin/linux-x86-32/activemq
--- apache-activemq-5.13.1/bin/linux-x86-32/activemq	2016-02-02 18:26:49.000000000 +0000
+++ apache-activemq-5.13.1.b/bin/linux-x86-32/activemq	2016-02-10 17:26:40.118057146 +0000
@@ -325,6 +325,12 @@
                 then
                     RUN_AS_GROUP=$RUN_AS_USER
                 fi
+                if [ ! -d "$PIDDIR" ]
+                then
+                   mkdir -p "$PIDDIR"
+                   chown $RUN_AS_USER:$RUN_AS_GROUP "$PIDDIR"
+                   chmod 0750 "$PIDDIR"
+                fi
                 touch $LOCKFILE
                 chown $RUN_AS_USER:$RUN_AS_GROUP $LOCKFILE
             fi
@@ -551,4 +557,4 @@
         ;;
 esac
 
-exit 0
\ No newline at end of file
+exit 0
diff -uNr apache-activemq-5.13.1/bin/linux-x86-64/activemq apache-activemq-5.13.1.b/bin/linux-x86-64/activemq
--- apache-activemq-5.13.1/bin/linux-x86-64/activemq	2016-02-02 18:26:49.000000000 +0000
+++ apache-activemq-5.13.1.b/bin/linux-x86-64/activemq	2016-02-10 17:26:56.502402290 +0000
@@ -325,6 +325,12 @@
                 then
                     RUN_AS_GROUP=$RUN_AS_USER
                 fi
+                if [ ! -d "$PIDDIR" ]
+                then
+                   mkdir -p "$PIDDIR"
+                   chown $RUN_AS_USER:$RUN_AS_GROUP "$PIDDIR"
+                   chmod 0750 "$PIDDIR"
+                fi
                 touch $LOCKFILE
                 chown $RUN_AS_USER:$RUN_AS_GROUP $LOCKFILE
             fi
@@ -551,4 +557,4 @@
         ;;
 esac
 
-exit 0
\ No newline at end of file
+exit 0
